File: single.cpp
#include<iostream>
using namespace std;

class Base
{
    public:
        int A,B;

        Base()
        {
            cout<<"Inside Base constructor\n";
        }
        ~Base()
        {
            cout<<"Inside Base destructor\n";
        }
        void fun()
        {
            cout<<"Inside Base fun\n";
        }
};

class Derived : public Base
{
    public :
        int X,Y;

        Derived()
        {
            cout<<"Inside derived constructor\n";
        }
        ~Derived()
        {
            cout<<"Inside derived destructor\n";
        }
        void gun()
        {
            cout<<"Inside gun of Derived\n";
        }
};

int main()
{
    Derived * ptr = NULL;

    ptr = new Derived;

    ptr->fun();
    ptr->gun();
    
    delete ptr;

    return 0;
}

File: Single.java
import java.util.*;

class Base
{
    public int A,B;
    public Base()
    {
        System.out.println("Base constructor");
    }
    public void fun()
    {
        System.out.println("Inside Base fun");
    }
    public void gun()
    {
        System.out.println("Inside Base gun");
    }
    public void fun(int iNo)
    {
        System.out.println("Inside Base fun with one integer");
    }

}

class Derived extends Base
{
    public int X,Y;

}

class Single
{
    public static void main (String arg[])
    {

    }
}